File: 7-kyu/strongn-strong-number-(special-numbers-series-#2)-/c++/solution.cpp
#include <string>
using namespace std;

int fac(int n)
{
    int ret = 1;
    for (int i = 1; i <= n; i++) {
        ret *= i;
    }
    return ret;
}

string strong_num (int number)
{
    int factorial = 0, n = number;
    while (number > 0) {
        factorial += fac(number % 10);
        number /= 10;
    }
    return factorial == n ? "STRONG!!!!" : "Not Strong !!";
}

File: 7-kyu/strongn-strong-number-(special-numbers-series-#2)-/python/solution.py
from functools import reduce
from operator import mul

def factorial(n):
    if n == 0: return 1
    if n <= 2: return n
    ret = reduce(mul, range(1, n + 1))
    return ret

def strong_num(number):
    number_lst = list(map(int, str(number)))
    fac_sum = sum(map(factorial, number_lst))
    return 'STRONG!!!!' if fac_sum == number else 'Not Strong !!'